### ISSUE TITLE
Facebook Internal Browser & Object.defineProperty

### ISSUE THREAD

**Issue by 3LOK** [polyfillpolyfill/polyfill-service#799]
When requesting 'fetch' using the Facebook Internal browser, polyfill.io decides to also send the Object.defineProperty polyfill (which is not needed as Facebook's browser has it). But the real problem starts due to a bug in Object.defineProperty implementation:

The following code works differently when using polyfill.io's implementation and chrome's implementation:

```
var x = {};
Object.defineProperty(x, 'hello', {value:'world'});
console.log(Object.keys(x));
```

In chrome it will print out [](empty array) while using the polyfill it will print out ['hello'](an array with the 'hello' element).

This causes a lot of problems...

**Comment by triblondon** [comment]
@3LOK thanks for this.  Can you provide the polyfill URL you are using and paste the contents of the comment from the top of the polyfill service output (remove `.min` if necessary)?

**Comment by JakeChampion** [comment]
Looks like there was some change in the definition of the function between es5 and es6

http://ecma-international.org/ecma-262/5.1/#sec-15.2.3.6
http://ecma-international.org/ecma-262/6.0/#sec-object.defineproperty

**Comment by triblondon** [comment]
@3LOC @oscar-b can either of you provide the UA string of the browser affected by this issue?

Also note that as a workaround, if the browsers concerned already have defineProperty, you can force the bundle to include protective feautre-detects by adding `flags=gated` to your polyfill.io querystring.

**Comment by JakeChampion** [comment]
This was fixed via #1147
